Sources/PackageLoading/PackageDescription4Loader.swift

```diff
@@ -298,51 +305,79 @@ extension PackageDependencyDescription.Requirement {
             let identifier = try json.get(String.self, forKey: "identifier")
             self = .exact(Version(string: identifier)!)
 
-        case "localPackage":
-            self = .localPackage
-
         default:
             throw InternalError("invalid dependency \(type)")
         }
     }
 }
 
 extension PackageDependencyDescription {
-    fileprivate init(v4 json: JSON, toolsVersion: ToolsVersion, packageLocation: String, fileSystem: FileSystem) throws {
-        let isBaseURLRemote = URL.scheme(packageLocation) != nil
-
-        func fixLocation(_ location: String, requirement: Requirement) throws -> String {
+    fileprivate init(
```

Comment:
👀 this is the key change, the logic to address identity resolution d location remapping (aka mirrors) moved to the constructor of the PackageDependencyDescription which is the easiest time and make the rest of the code simpler can it assume the dependency identity and location is correct